### DISCUSSION TITLE
Wrong floatingStyles when used in jest + rtl environment

### DISCUSSION THREAD

**Discussion by R-Oscar** [floating-ui/floating-ui#2627] [Q&A] [answered]
Hey everyone

I've encountered a problem while testing the component that uses useFloating hook
It is that in the test file coordinates are zeros, while in an actual App, floating element is positioned correctly

I've made a simple reproduction in my repo: https://github.com/R-Oscar/floating-ui-jest-test
1. `npm i`
2. `npm run dev`, open DevTools and see that styles are applied correctly: `transform: translate(97px, 520px);`
3. `npm run test` and look for console.log output: `transform: 'translate(0px, 0px)'`

I also did this in my test case: `await act(async () => {}); // Flush microtasks.`, but as you can see it didn't help
Is there anything I can do in order to fix this behavior? Thanks in advance!

**Comment by atomiks** [comment, marked as answer]
JSDOM can't render elements to measure in its virtual env, so measured sizes are always `0`. 

Therefore you can't assert the sizes or location of the elements inside such an env - this requires Playwright or similar e2e tests.

**Reply by R-Oscar** [reply]
Oh I see, thank you for the response
